.gitlab-ci.yml

```diff
@@ -29,22 +29,56 @@ variables:
     paths:
       - artifacts/
 
-test-linux:
+test-audit:
   stage:                           test
-  variables:
-    RUN_TESTS:                     all
   script:
-    - scripts/gitlab/test-all.sh
+    - set -e
+    - set -u
+    - cargo audit
+  tags:
+    - linux-docker
+
+test-validate-chainspecs:
+  stage:                           test
+  script:
+    - ./scripts/gitlab/validate_chainspecs.sh
     - sccache -s
   tags:
     - linux-docker
 
-test-audit:
+cargo-check:
   stage:                           test
   script:
-    - set -e
-    - set -u
-    - cargo audit
+    - ./scripts/gitlab/cargo_check.sh
+    - sccache -s
+  tags:
+    - linux-docker
+
+test-rust-changes:
+  stage:                           test
+  script:
+    - ./scripts/gitlab/rust_changes.sh
```

Comment:
How can it end the pipeline with success when no relevant files have changed?

Reply:
not sure yet, up to test it

Reply:
but it's not supposed to mark the pipeline as successful, previously it stopped tests from being run

Reply:
well, you should not be able to merge if the test are not successful.